src/main/java/uk/gov/hmcts/ccd/data/casedetails/query/UserAuthorisationSecurity.java

```diff
@@ -0,0 +1,24 @@
+package uk.gov.hmcts.ccd.data.casedetails.query;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.infrastructure.user.UserAuthorisation;
+import uk.gov.hmcts.ccd.infrastructure.user.UserAuthorisation.AccessLevel;
+
+@Component
+class UserAuthorisationSecurity {
+
+    private final UserAuthorisation userAuthorisation;
+
+    @Autowired
+    UserAuthorisationSecurity(UserAuthorisation userAuthorisation) {
+        this.userAuthorisation = userAuthorisation;
+    }
+
+    <T> CaseDetailsQueryBuilder<T> secure(CaseDetailsQueryBuilder<T> builder) {
+        if (AccessLevel.GRANTED.equals(userAuthorisation.getAccessLevel())) {
```

Comment:
could we have the query builder itself handling this instead of introducing  a new component?

Reply:
I think that would be giving too much responsibility to the QueryBuilder.
Also, the QueryBuilder is quite hard to unit tests given it's using the Criteria API. Keeping the security logic separated gives the abstraction necessary to easily unit test the security logic.